src/main/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriber.java

```diff
@@ -73,10 +75,14 @@ protected boolean isApplicable(@CheckForNull final Item item) {
     @Override
     @SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE", justification = "Return value of parseEventPayload method is safe to cast.")
     protected void onEvent(final GHSubscriberEvent event) {
-        String payload = event.getPayload();
+        final String payload = event.getPayload();
+        final String branchName;
+
         try {
             GHEventPayload.CheckRun checkRun = GitHub.offline().parseEventPayload(new StringReader(payload), GHEventPayload.CheckRun.class);
-
+            JSONObject payloadJSON = new JSONObject(payload);
+            branchName = payloadJSON.getJSONObject("check_run").getJSONObject("check_suite").getString("head_branch");
```

Comment:
looks like this can be null:
https://github.com/jenkins-infra/helpdesk/issues/2738#issuecomment-1013648010

Probably when using the pr-merge strategy when builds aren't up to date